lib/Dialect/ESI/ESIDialect.cpp

```diff
@@ -82,6 +82,72 @@ void circt::esi::findValidReadySignals(
   }
 }
 
+/// Given a list of logical port names, find the data/valid/ready port triples.
+void circt::esi::resolvePortNames(
```

Comment:
Seems like a lot of duplication here with `findValidReadySignals`. How do they differ? It seems like `findValidReadySignals` is using its heuristic to determine how data/valid/ready are named, whereas `resolvePortNames` lets the user specify the base name to look for?

To me it seems like `resolvePortNames` was a copied and tweaked, and they can be more unified. Could `resolvePortNames` call into `findValidReadySignals`? Is there a use-case (besides the tests) for `findValidReadySignals` if `resolvePortNames` exists as it does in this PR? At the least, it seems like a helper could be pulled out for "Find a port named like this", which they could both use, if it even makes sense to have the two separate APIs.

Reply:
Yeah, I was thinking about that while I was copying the code. My thought at the time is that the details are too different to make the additional complexity worth while. I've factored the common stuff out, and it's probably worth the additional complexity but not by much IMO.

As for the use case of `findValidReadySignals`, I think there is still one in wrapping handshake and other cases where you want to do the same thing on all of the LI ports (e.g. connect them all via cosim). In cases where you need to do different things, you kinda need to know the port names. There may be some exceptions based on the type.

Reply:
Thanks for doing the refactoring. It looks good to me.